src/ol/source/VectorEventType.js

```diff
@@ -34,4 +34,25 @@ export default {
    * @api
    */
   REMOVEFEATURE: 'removefeature',
+
+  /**
+   * Triggered when features starts loading.
+   * @event module:ol/source/Vector.VectorSourceEvent#featureloadstart
+   * @api
+   */
+  FEATURELOADSTART: 'featureloadstart',
```

Comment:
Maybe just a matter of taste, but I think the event names could be shorter: `loadstart`, `loadend`, `loaderror`.

Reply:
I made it this way, because it is similar to the `tileloadstart` and `imageloadstart`. But I don't see why these have to be named differently in the first place. We could also introduce these events in `source/Source` if we implemented them in `source/UTFGrid` as well. They would not be fired in `source/Tile` and `source/Source` then, but those are abstract anyways.

Reply:
I noticed that the other load events always contain the image or tile. This way the names make sense and they cannot be generalized that easily.

It could make sense to provide the loaded features in the `loadend` event for other than the default loading strategy.

Reply:
The reason why I started thinking about it was because we're loading `features`, not a `feature`. `featuresloadstart` would be a long event name, but it would be the correct name. And yes, it might make sense to provide the loaded features in the `featuresloadend` event.